packages/plugin-loyalties-api/src/models/Vouchers.ts

```
@@ -159,12 +162,74 @@ export const loadVoucherClass = (models: IModels, subdomain: string) => {
+        throw new Error("The campaign has ended and the voucher is expired");
+      }
+
+      return voucherCampaign;
     }
 
-    public static async confirmVoucherSale({ checkInfo }) {
-      return confirmVoucherSale(models, checkInfo);
+    public static async redeemVoucher({ ownerType, usageInfo }) {
+      const { ownerId, voucherId } = usageInfo || {};
+
+      const isValid = await this.checkVoucher({
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)